src/connector/with_options_source.yaml

```diff
@@ -397,6 +428,33 @@ NexmarkProperties:
     field_type: usize
     required: false
     default: None
+OpendalS3Properties:
+  fields:
+  - name: s3.region_name
+    field_type: String
+    required: true
+  - name: s3.bucket_name
+    field_type: String
+    required: true
+  - name: match_pattern
+    field_type: String
+    required: false
+    default: Default::default
+  - name: s3.credentials.access
+    field_type: String
+    required: false
+    default: Default::default
+  - name: s3.credentials.secret
+    field_type: String
+    required: false
+    default: Default::default
+  - name: s3.endpoint_url
+    field_type: String
+    required: false
+  - name: s3.assume_role
+    field_type: String
+    required: false
+    default: Default::default
```

Comment:
I'm currently not sure whether this is correct. `OpendalS3Properties` do not use serde directly.

Reply:
Update: this concern should be fixed after #14190